feat: initial commit for DA

maheshwarishikha opened the pull request:
### Description

VPC DA - fully configurable variant

Issue: https://github.ibm.com/GoldenEye/issues/issues/12217

<!--- Replace this text with a summary of the changes in this PR. Include why the changes are needed and context about the changes. List required dependencies. If there is a Git issue for the change, please link to it. --->

### Release required?
<!--- Identify the type of release. For information about the changes in a semantic versioning release, see [Release versioning](https://terraform-ibm-modules.github.io/documentation/#/versioning). --->

- [ ] No release
- [ ] Patch release (`x.x.X`)
- [ ] Minor release (`x.X.x`)
- [x] Major release (`X.x.x`)

##### Release notes content

- Fully configurable variant
- Adds vpn_gateways feature in base module

<!--- If a release is required, replace this text with information that users need to know about the release. Write the release notes to help users understand the changes, and include information about how to update from the previous version.

Your notes help the merger write the commit message for the PR that is published in the release notes for the module. --->

### Run the pipeline

If the CI pipeline doesn't run when you create the PR, the PR requires a user with GitHub collaborators access to run the pipeline.

Run the CI pipeline when the PR is ready for review and you expect tests to pass. Add a comment to the PR with the following text:

```
/run pipeline
```

### Checklist for reviewers

- [ ] If relevant, a test for the change is included or updated with this PR.
- [ ] If relevant, documentation for the change is included or updated with this PR.

### For mergers

- Use a conventional commit message to set the release level. Follow the [guidelines](https://terraform-ibm-modules.github.io/documentation/#/merging.md).
- Include information that users need to know about the PR in the commit message. The commit message becomes part of the GitHub release notes.
- Use the **Squash and merge** option.

ocofaigh commented on solutions/simple/main.tf:
```diff
@@ -0,0 +1,128 @@
+
+# parse COS details from the existing COS instance CRN
+module "existing_cos_crn_parser" {
+  count   = var.existing_cos_instance_crn != null ? 1 : 0
```
using a CRN around a count can lead into issues where count is not known before apply. I suspect however we will never hit this in a DA unless someone tries to consume it as a module where they also create the COS instance in the same state file

ocofaigh commented on solutions/simple/main.tf:
```diff
@@ -0,0 +1,128 @@
+}
+
+module "cos_buckets" {
+  count          = var.enable_vpc_flow_logs ? 1 : 0
```
are we supporting user bringing there own bucket?

maheshwarishikha replied:
No, in simple variation...I planned to skip supporting existing storage bucket. So user will bring the COS instance and a new bucket will always be created.

We can discuss...currently it is implemented like this.

ocofaigh commented on solutions/simple/variables.tf:
```diff
@@ -0,0 +1,400 @@
+variable "resource_group_name" {
+  type        = string
+  description = "The name of a new or an existing resource group to provision the watsonx.ai resources. If a prefix input variable is specified, the prefix is added to the name in the `<prefix>-<name>` format."
+  default     = "Default"
```
If we want to be consistent with other DAs, this should have no default value

maheshwarishikha replied:
yes, that was by mistake. Modified.

ocofaigh commented on solutions/simple/variables.tf:
```diff
@@ -0,0 +1,400 @@
+
+variable "vpc_name" {
+  default     = "simple"
+  description = "Name of the VPC."
```
description needs to mention the prefix variable

maheshwarishikha replied:
updated.

ocofaigh commented on solutions/simple/variables.tf:
```diff
@@ -0,0 +1,400 @@
+
+variable "access_tags" {
+  type        = list(string)
+  description = "A list of access tags to apply to the VPC resources created by the module. For more information, see https://cloud.ibm.com/docs/account?topic=account-access-tags-tutorial."
```
Its not a module, so don't use the word "module" in descrption

ocofaigh commented on solutions/simple/variables.tf:
```diff
@@ -0,0 +1,400 @@
+##############################################################################
+
+variable "subnets" {
+  description = "List of subnets for the vpc. For each item in each array, a subnet will be created. Items can be either CIDR blocks or total ipv4 addressess. Public gateways will be enabled only in zones where a gateway has been created"
```
We should add a supporting markdown doc

ocofaigh commented on solutions/simple/variables.tf:
```diff
@@ -0,0 +1,400 @@
+##############################################################################
+
+variable "network_acls" {
+  description = "The list of ACLs to create. Provide at least one rule for each ACL."
```
We should add a supporting markdown doc

maheshwarishikha replied:
will add, once code is completed.

maheshwarishikha replied:
added

ocofaigh commented on solutions/simple/variables.tf:
```diff
@@ -0,0 +1,400 @@
+  default     = false
+}
+
+variable "create_authorization_policy_vpc_to_cos" {
```
keep consistent with other DAs - use the `skip_` naming convention

maheshwarishikha replied:
done

ocofaigh commented on solutions/simple/variables.tf:
```diff
@@ -0,0 +1,400 @@
+}
+
+variable "existing_cos_instance_crn" {
+  description = "GUID of the COS instance to create Flow log collector"
```
GUID -> CRN

maheshwarishikha replied:
updated.

ocofaigh commented on solutions/simple/version.tf:
```diff
@@ -0,0 +1,10 @@
+terraform {
+  required_version = ">= 1.3.0"
```
you should bump this to >= 1.9.0 and use cross variable validation. There are many combinations / permutations so we badly need to use that kind of validation in the variables.

maheshwarishikha replied:
done

ocofaigh reviewed (commented):
I know it is still under development, but I left a few comments to make sure we address them. We should talk about the variation name. `simple` is probably not what we will go with

maheshwarishikha commented:
> I know it is still under development, but I left a few comments to make sure we address them. We should talk about the variation name. `simple` is probably not what we will go with

I created this draft PR with the intent of early review. Thanks for this. Yes, name will be changed once we decide.

maheshwarishikha commented:
Currently,
- this simple variation does not support `existing_cos_kms_key_crn` scenario.
- it does not support the existing COS buckets. 
More secured version can have both of these. Let me know your thoughts.

ocofaigh commented on solutions/simple/version.tf:
```diff
@@ -0,0 +1,10 @@
+    # Use "greater than or equal to" range in modules
+    ibm = {
+      source  = "IBM-Cloud/ibm"
+      version = ">= 1.59.0, < 2.0.0"
```
Lock into exact version for a DA

maheshwarishikha replied:
addressed

ocofaigh commented on solutions/simple/version.tf:
```diff
@@ -0,0 +1,10 @@
+terraform {
+  required_version = ">= 1.9.0"
+  required_providers {
+    # Use "greater than or equal to" range in modules
```
`# Lock DA into an exact provider version - renovate automation will keep it updated`

maheshwarishikha replied:
added

ocofaigh commented on solutions/simple/variables.tf:
```diff
@@ -0,0 +1,399 @@
+}
+
+variable "default_routing_table_name" {
+  description = "OPTIONAL - Name of the Default Routing Table. If null, a name will be automatically generated"
```
The guidance from docs team is not to add `OPTIONAL` in the variable description. In projects it will show in the optional tab, so no need to duplicate that

maheshwarishikha replied:
done

ocofaigh reviewed (changes requested):
We really need to make use of cross variable validation in the variables.tf file. For example, if person sets `kms_encryption_enabled_bucket` or `enable_vpc_flow_logs` then other inputs become required. 

Also question - should we support creating VPEs? Or are we planning to have a standalone VPE DA? (cc @vburckhardt )

ocofaigh commented:
@maheshwarishikha FYI these are the variation names we wil go with:

1. label: `Fully configurable`, name: `fully-configurable`
2. label: Security-enforced, name: `security-enforced`

ocofaigh commented on solutions/fully-configurable/variables.tf:
```diff
@@ -0,0 +1,473 @@
+}
+
+variable "vpc_name" {
+  default     = "simple"
```
we probably need a better default than this? Perhaps it should be "vpc"? So you would end up with "dev-vpc" when using prefix?

maheshwarishikha replied:
updated

ocofaigh commented on solutions/fully-configurable/variables.tf:
```diff
@@ -0,0 +1,492 @@
+}
+
+variable "kms_encryption_enabled_bucket" {
+  description = "Set to true if Cloud Object Storage bucket needs to be KMS encryption enabled."
```
Set to true to encrypt the Cloud Object Storage Flow Logs bucket with a KMS key. If set to true, a value must be passed for `existing_kms_key_crn` (to use that key) or `existing_kms_instance_crn` (to create a new key). Value cannot be set to true if `enable_vpc_flow_logs` is set to false.

maheshwarishikha replied:
updated

ocofaigh commented on solutions/fully-configurable/variables.tf:
```diff
@@ -0,0 +1,492 @@
+# KMS
+###############################################################################################################
+
+variable "existing_kms_key_crn" {
```
I actually suggest you name this `existing_flow_logs_bucket_kms_key_crn` so it is clear what the key is used for. Maybe some day there be something else in this DA that will need its own KMS key, so better to future proof it

maheshwarishikha replied:
updated

ocofaigh commented on solutions/fully-configurable/variables.tf:
```diff
@@ -0,0 +1,492 @@
+variable "provider_visibility" {
+  description = "Set the visibility value for the IBM terraform provider. Supported values are `public`, `private`, `public-and-private`. [Learn more](https://registry.terraform.io/providers/IBM-Cloud/ibm/latest/docs/guides/custom-service-endpoints)."
+  type        = string
+  default     = "public"
```
set this to private

maheshwarishikha replied:
done

ocofaigh commented on solutions/fully-configurable/variables.tf:
```diff
@@ -0,0 +1,492 @@
+variable "prefix" {
+  type        = string
+  nullable    = true
+  description = "Prefix to add to all the resources created by this solution. To not use any prefix value, you can set this value to `null` or an empty string."
```
since this DA support creating a COS bucket, you should add extra validation on the value. See https://github.com/terraform-ibm-modules/terraform-ibm-scc-da/blob/49ba987b2679e8e8bae0af451922ec525252fe92/solutions/instances/variables.tf#L33-L40

maheshwarishikha replied:
added

ocofaigh commented on solutions/fully-configurable/variables.tf:
```diff
@@ -0,0 +1,492 @@
+  type        = string
+}
+
+variable "resource_tags" {
```
different vpc resource support tags (e.g. a VPC can have different tags to subnets). I think this needs to be broken down. VPC tags variable should be explicitly named `vpc_resource_tags`

maheshwarishikha replied:
As discussed, subnet_tags has been added explicitly.

ocofaigh commented on solutions/fully-configurable/variables.tf:
```diff
@@ -0,0 +1,492 @@
+
+variable "subnets" {
+  description = "List of subnets for the vpc. For each item in each array, a subnet will be created. Items can be either CIDR blocks or total ipv4 addressess. Public gateways will be enabled only in zones where a gateway has been created"
+  type = object({
```
I suggest to add a `subnet_resource_tags` attribute so subnets can individually tagged

maheshwarishikha replied:
should we make this change only in DA or in module?

ocofaigh commented on solutions/fully-configurable/variables.tf:
```diff
@@ -0,0 +1,492 @@
+variable "clean_default_sg_acl" {
+  description = "Remove all rules from the default VPC security group and VPC ACL (less permissive)"
+  type        = bool
+  default     = false
```
hmm, should this default to secure by design (aka true) ?

maheshwarishikha replied:
I think, we can make it true for security-enforced variant. what say?

ocofaigh replied:
I think if set to false, it will fail one of the SCC rules as the rules are wide open. But if we set it to false, anything that is using the default ACL or SG will fail as it will have no rules, so no communication can occur unless user sets up an alternative ACL / SG. @toddgiguere any thoughts?

toddgiguere replied:
I think having the default as "true" is only an issue when the subnets have the default ACL attached. I think that might be impossible in this case? If so then the default ACLs are orphans anyway (not attached to any subnets) so if they are empty (even accidentally) it won't matter? I'm thinking from ACL standpoint it might be safe to default to `true`.

Security Group is a little more tricky. By default it is (probably) not attached to anything in this module, but it is still out there and available, but I'm not sure how many would use the default sg, rather than create a new one.

To be honest I don't really see this as a "security enforcement", because these are not used or attached to anything so setting this to "true" isn't really affecting any traffic. We really added this because SCC is not smart enough to figure out that the ACLs are not attached to anything, and flags them as "too open" anyway.

maheshwarishikha replied:
Got the point. thanks. 
Made it true.

ocofaigh commented on solutions/fully-configurable/variables.tf:
```diff
@@ -0,0 +1,492 @@
+}
+
+variable "existing_cos_instance_crn" {
+  description = "CRN of the existing COS instance. It is required to create the bucket used for flow logs."
```
You should mention it is only required if `enable_vpc_flow_logs` is set to true in the description

maheshwarishikha replied:
added

ocofaigh commented on solutions/fully-configurable/variables.tf:
```diff
@@ -0,0 +1,492 @@
+  }
+}
+
+variable "cos_bucket_name" {
```
perhaps call it `flow_logs_new_bucket_name` so its clear that its a new bucket that will be created for flow logs

maheshwarishikha replied:
Modified as `flow_logs_cos_bucket_name` , description says bucket will be created.

ocofaigh commented on solutions/fully-configurable/variables.tf:
```diff
@@ -0,0 +1,492 @@
+variable "cos_bucket_name" {
+  description = "Name of the Cloud Object Storage bucket to be created collect VPC flow logs."
+  type        = string
+  default     = "cos-bucket"
```
maybe rename this to flow-logs-bucket

maheshwarishikha replied:
done

ocofaigh commented on solutions/fully-configurable/variables.tf:
```diff
@@ -0,0 +1,492 @@
+variable "management_endpoint_type_for_bucket" {
+  description = "The type of endpoint for the IBM Terraform provider to use to manage Cloud Object Storage buckets (`public`, `private`, or `direct`). If you are using a private endpoint, make sure that you enable virtual routing and forwarding (VRF) in your account, and that the Terraform runtime can access the IBM Cloud Private network."
+  type        = string
+  default     = "public"
```
default this to direct

maheshwarishikha replied:
done

ocofaigh commented on solutions/fully-configurable/variables.tf:
```diff
@@ -0,0 +1,492 @@
+variable "kms_endpoint_type" {
+  type        = string
+  description = "The type of endpoint to use for communicating with the Key Protect instance. Possible values: `public`, `private`. Applies only if `existing_cos_kms_key_crn` is not specified."
+  default     = "public"
```
private

maheshwarishikha replied:
done

ocofaigh commented on solutions/fully-configurable/variables.tf:
```diff
@@ -0,0 +1,492 @@
+  }
+}
+
+variable "kms_key_ring_name" {
```
should we be more explicit? `flow_logs_bucket_kms_key_ring_name` ?

maheshwarishikha replied:
The customers create separate keys for different purposes. Is the creation of different key-rings recommended?

ocofaigh replied:
I think the best practise for key rings is to split them by functional area. So maybe this is fine to leave as `kms_key_ring_name` and perhaps we default it to something like `vpc-key-ring` ?

ocofaigh commented on solutions/fully-configurable/variables.tf:
```diff
@@ -0,0 +1,492 @@
+
+variable "kms_key_ring_name" {
+  type        = string
+  default     = "cos-key-ring"
```
flow-logs-bucket-key-ring

ocofaigh replied:
as per above [comment](https://github.com/terraform-ibm-modules/terraform-ibm-landing-zone-vpc/pull/915#discussion_r1991378922), maybe go with `vpc-key-ring` ?

ocofaigh commented on solutions/fully-configurable/variables.tf:
```diff
@@ -0,0 +1,492 @@
+  description = "The name of the key ring to create for the Cloud Object Storage bucket key. If an existing key is used, this variable is not required. If the prefix input variable is passed, the name of the key ring is prefixed to the value in the `<prefix>-value` format."
+}
+
+variable "kms_key_name" {
```
should we be more explicit? flow_logs_bucket_kms_key_name ?

ocofaigh commented on solutions/fully-configurable/variables.tf:
```diff
@@ -0,0 +1,492 @@
+
+variable "kms_key_name" {
+  type        = string
+  default     = "cos-key"
```
flow-logs-bucket-key

ocofaigh commented:
Should we be supporting existing VPC @vburckhardt ?

maheshwarishikha commented on solutions/fully-configurable/main.tf:
```diff
@@ -0,0 +1,136 @@
+    storage_class                 = var.cos_bucket_class
+    resource_instance_id          = var.existing_cos_instance_crn
+    region_location               = var.region
+    force_delete                  = true
```
exposed via the variable

maheshwarishikha commented:
**default inbound rules:**

<img width="1103" alt="image" src="https://github.com/user-attachments/assets/73e82c44-973e-45e3-a335-97b5c0068f71" />

**default outbound rules:**

<img width="1102" alt="image" src="https://github.com/user-attachments/assets/7de0b198-1e36-4b67-b647-ff597d2a64f4" />

maheshwarishikha commented:
@ocofaigh @toddgiguere Please review this PR.

Few points/questions :
1. Have added all the required features except separate resource tags functionality for vpc, subnets, vpn gateways etc. Working on this. One question - as we plan to change variable names in module, will it be a breaking change for landing zone?

2. I have added VPE support but it is being attached with default security groups. Current code does not support creating more security groups as discussed before.

3. After adding VPE, `tf plan` shows `update in-place`. And `tf apply` gave error as:
```
Error: Error Deleting Security Group Target for this endpoint gateway : The specified endpoint gateway is not attached to any other security groups.
```
Is it a known issue? Seems to me a provider issue.

4. Docs for complex objects will be added soon.

ocofaigh commented:
> 1. Have added all the required features except separate resource tags functionality for vpc, subnets, vpn gateways etc. Working on this. One question - as we plan to change variable names in module, will it be a breaking change for landing zone?

^ Renaming variables is not a breaking change. Its a minor version change with release notes that clearly state the renamed vars. See this [doc](https://terraform-ibm-modules.github.io/documentation/#/versioning?id=major-version-changes) for breaking change criteria

> 2. I have added VPE support but it is being attached with default security groups. Current code does not support creating more security groups as discussed before.

^ Remind me why we are not supporting creation of more security groups? I would of thought this needs to be a feature of a VPC DA?

> 3. After adding VPE, tf plan shows update in-place. And tf apply gave error

^ How do we handle that provider bug in any of our other modules / examples that create VPE? Can we follow up with Ujjwal Kumar ?

> 4. Docs for complex objects will be added soon.

^ ok

> 5. Currently I have not added the following for cos buckets. Let me know if need to add this too.

^ This COS bucket will be used for VPC flow logs. So by default it should create an audit type bucket (retention enabled by default etc)

maheshwarishikha commented:
> This COS bucket will be used for VPC flow logs. So by default it should create an audit type bucket (retention enabled by default etc)

Added archive, expire, retention rules and object versioning.

> Regarding renaming of variables - tags related.

We have tags for vpc, subnet, flow logs, vpe gateway, vpn gateway, public gateway - should we really plan to add resource_tags for each?? Currently we use `tags` variable for each.

> Remind me why we are not supporting creation of more security groups? I would of thought this needs to be a feature of a VPC DA?

Previously we thought that it is an advanced feature, but that discussion was for quickstart. Added it now.

> How do we handle that provider bug in any of our other modules / examples that create VPE? 

Having default value as `null` instead of `[]` is helping. Made the same change in code.

maheshwarishikha commented:
/run pipeline

maheshwarishikha commented:
FullyConfigurable upgrade test is not there in main branch, hence skipping upgrade test.

maheshwarishikha commented:
/run pipeline

ocofaigh commented on solutions/fully-configurable/variables.tf:
```diff
@@ -0,0 +1,679 @@
+  }
+}
+
+variable "force_delete" {
```
Don't expose this - it will prevent destroy. See https://ibm-cloudplatform.slack.com/archives/CG06FQ31P/p1741861817152299

maheshwarishikha replied:
sure, removed.

ocofaigh commented on solutions/fully-configurable/variables.tf:
```diff
@@ -0,0 +1,679 @@
+variable "flow_logs_cos_bucket_archive_days" {
+  description = "The number of days before the `archive_type` rule action takes effect for the flow logs cloud object storage bucket."
+  type        = number
+  default     = 20
```
curious where this number came from?

maheshwarishikha replied:
I took this number from the default value defined in module - https://github.com/terraform-ibm-modules/terraform-ibm-cos/blob/main/modules/buckets/variables.tf#L36

ocofaigh commented on solutions/fully-configurable/variables.tf:
```diff
@@ -0,0 +1,679 @@
+variable "existing_flow_logs_bucket_kms_key_crn" {
+  type        = string
+  default     = null
+  description = "The CRN of the existing root key of key management service (KMS) that is used to encrypt the flow logs Cloud Object Storage bucket."
```
the description should mention to use `existing_kms_instance_crn` to create a new key

maheshwarishikha replied:
updated.

ocofaigh commented on solutions/fully-configurable/variables.tf:
```diff
@@ -0,0 +1,679 @@
+# VPE Gateways
+##############################################################################
+
+variable "cloud_services" {
```
suggest to prefix this with `vpe_gateway_`

maheshwarishikha replied:
updated

ocofaigh commented on solutions/fully-configurable/variables.tf:
```diff
@@ -0,0 +1,679 @@
+  default = []
+}
+
+variable "cloud_service_by_crn" {
```
suggest to prefix this with `vpe_gateway_`

maheshwarishikha replied:
updated

ocofaigh commented on solutions/fully-configurable/variables.tf:
```diff
@@ -0,0 +1,679 @@
+##############################################################################
+
+variable "cloud_services" {
+  description = "The list of cloud services used to create endpoint gateways. If `vpe_name` is not specified in the list, VPE names are created in the format `<prefix>-<vpc_name>-<service_name>`."
```
The description needs to point to a helper markdown doc

maheshwarishikha replied:
added.

ocofaigh commented on solutions/fully-configurable/variables.tf:
```diff
@@ -0,0 +1,679 @@
+}
+
+variable "cloud_service_by_crn" {
+  description = "The list of cloud service CRNs used to create endpoint gateways. Use this list to identify services that are not supported by service name in the `cloud_services` variable. For a list of supported services, see [VPE-enabled services](https://cloud.ibm.com/docs/vpc?topic=vpc-vpe-supported-services). If `service_name` is not specified, the CRN is used to find the name. If `vpe_name ...
```
The description needs to point to a helper markdown doc

maheshwarishikha replied:
added

ocofaigh commented on solutions/fully-configurable/variables.tf:
```diff
@@ -0,0 +1,679 @@
+  default = []
+}
+
+variable "vpe_service_endpoints" {
```
suggest to prefix this with `vpe_gateway_`

maheshwarishikha replied:
updated

ocofaigh reviewed (changes requested):
a few comments

maheshwarishikha commented:
> We have tags for vpc, subnet, flow logs, vpe gateway, vpn gateway, public gateway - should we really plan to add resource_tags for each?? Currently we use tags variable for each.

Discussed with Conall, the plan to do the following:
Use `var.tags` for all VPC related resources created by the DA. For extra subnet related tags, just add 1 more attribute in var.subnets to support adding different tags per subnet zone

maheshwarishikha commented on variables.tf:
```diff
@@ -718,3 +718,28 @@ variable "dns_records" {
+# VPN Gateways
+##############################################################################
+
+variable "enable_vpn_gateways" {
```
thanks for the suggestion, updated code.

maheshwarishikha commented:
/run pipeline

maheshwarishikha commented:
/run pipeline

maheshwarishikha commented:
/run pipeline

maheshwarishikha commented:
/run pipeline

Khuzaima05 commented:
/run pipeline

Khuzaima05 commented:
/run pipeline

Khuzaima05 commented on ibm_catalog.json:
```diff
@@ -0,0 +1,533 @@
+                  "reference_version": true
+                },
+                {
+                  "dependency_input": "use_existing_resource_group",
```
@rajatagarwal-ibm use_existing_resource_group variable is part of kms add-on , here we are passing value to that variable.

rajatagarwal-ibm replied:
you're right, my bad.

maheshwarishikha commented on solutions/fully-configurable/main.tf:
```diff
@@ -0,0 +1,245 @@
+locals {
+  cos_instance_guid                        = var.existing_cos_instance_crn != null ? module.existing_cos_crn_parser[0].service_instance : null
+  cos_account_id                           = var.existing_cos_instance_crn != null ? module.existing_cos_crn_parser[0].account_id : null
+  bucket_name                              = "${local.prefix}${var.flow_logs_cos_bucket_name}"
```
This is correct. Its a new change `prefix` related. Now `local.prefix` has `-` already.

maheshwarishikha commented on solutions/fully-configurable/main.tf:
```diff
@@ -0,0 +1,245 @@
+    kms_encryption_enabled        = var.kms_encryption_enabled_bucket
+    kms_guid                      = local.kms_guid
+    kms_key_crn                   = local.cos_kms_key_crn
+    skip_iam_authorization_policy = local.create_cross_account_cos_kms_auth_policy || var.skip_cos_kms_iam_auth_policy
```
`local.create_cross_account_cos_kms_auth_policy` will be true incase of cross account and if skip** is true then it means Buckets module wont create the policy. So I think it is fine.

maheshwarishikha commented on solutions/fully-configurable/variables.tf:
```diff
@@ -0,0 +1,694 @@
+variable "management_endpoint_type_for_bucket" {
+  description = "The type of endpoint for the IBM Terraform provider to use to manage Cloud Object Storage buckets (`public`, `private`, or `direct`). If you are using a private endpoint, make sure that you enable virtual routing and forwarding (VRF) in your account, and that the Terraform runtime can access the IBM Cloud Private network."
+  type        = string
+  default     = "direct"
```
It is as per the suggestion provided by Conall. For DAs, we are keeping it direct everywhere.

Khuzaima05 commented on ibm_catalog.json:
```diff
@@ -0,0 +1,533 @@
+          "working_directory": "solutions/fully-configurable",
+          "architecture": {
+            "descriptions": "This architecture creates a VPC.",
+            "features": [
```
Updated the architecture description and acronyms. All add-on-related features are already covered at the top-level feature, so we are not including them in the flavor-specific features to avoid duplication.

maheshwarishikha replied:
We may be doing the same for other DAs going forward. 
Refer - https://github.com/terraform-ibm-modules/terraform-ibm-scc/blob/main/ibm_catalog.json#L386

Khuzaima05 commented on .catalog-onboard-pipeline.yaml:
```diff
@@ -0,0 +1,14 @@
+        mark_ready: true
+        install_type: fullstack
+        scc:
+          instance_id: 1c7d5f78-9262-44c3-b779-b28fe4d88c37
```
done

Khuzaima05 commented on ibm_catalog.json:
```diff
@@ -0,0 +1,533 @@
+        "solution"
+      ],
+      "short_description": "Provisions a VPC network on IBM Cloud",
+      "long_description": "The VPC deployable architecture deploys a Virtual Private Cloud (VPC) infrastructure without any compute resources, such as Virtual Server Instances (VSI) or Red Hat OpenShift clusters.",
```
Done

Khuzaima05 commented on ibm_catalog.json:
```diff
@@ -0,0 +1,533 @@
+        "infrastructure as code",
+        "solution"
+      ],
+      "short_description": "Provisions a VPC network on IBM Cloud",
```
Done

Khuzaima05 commented on .catalog-onboard-pipeline.yaml:
```diff
@@ -0,0 +1,14 @@
+  - name: deploy-arch-ibm-vpc
+    kind: solution
+    catalog_id: f64499c8-eb50-4985-bf91-29f9e605a433
+    offering_id: ae881fc1-273f-472e-8437-99099a6a753b
```
updated id

maheshwarishikha replied:
updated

Khuzaima05 commented on ibm_catalog.json:
```diff
@@ -0,0 +1,533 @@
+                  "url": "https://raw.githubusercontent.com/terraform-ibm-modules/terraform-ibm-landing-zone-vpc/main/reference-architecture/deployable-architecture-vpc.svg",
+                  "type": "image/svg+xml"
+                },
+                "description": "This architecture creates a VPC."
```
updated

maheshwarishikha commented on solutions/fully-configurable/variables.tf:
```diff
@@ -0,0 +1,694 @@
+variable "flow_logs_cos_bucket_maximum_retention_days" {
+  description = "The maximum number of days that an object can be kept unmodified in the flow logs cloud object storage."
+  type        = number
+  default     = 350
```
For retention...thats the value we keep it. I'll check docs more on it. We use these values everytime.
https://ibm-cloudplatform.slack.com/archives/C0117873D0S/p1716805861637559

Khuzaima05 commented on solutions/fully-configurable/variables.tf:
```diff
@@ -0,0 +1,694 @@
+variable "flow_logs_cos_bucket_archive_type" {
+  description = "The storage class or archive type you want the object to transition to in the flow logs cloud object storage bucket."
+  type        = string
+  default     = "Glacier"
```
Done

Khuzaima05 commented on tests/pr_test.go:
```diff
@@ -201,3 +203,93 @@ func TestRunVpcWithDnsExample(t *testing.T) {
+		TemplateFolder:         fullyConfigFlavorDir,
+		Tags:                   []string{"vpc-da-test"},
+		DeleteWorkspaceOnFail:  false,
+		WaitJobCompleteMinutes: 60,
```
increased to 120

rajatagarwal-ibm commented on ibm_catalog.json:
```diff
@@ -0,0 +1,543 @@
+        "solution"
+      ],
+      "short_description": "Automates VPC deployment on IBM Cloud, offering full configurability and flexibility for diverse workloads.",
+      "long_description": "The VPC deployable architecture deploys a Virtual Private Cloud (VPC) infrastructure without any compute resources, such as Virtual Server Instances (VSI) or Red Hat OpenShift clusters.This is an experimental tile and not suitable for production workloads. Stay here if you want to try an experimental version with the Optional and swappable components capability.",
```
can you please add the URLs pointing to the optional and swappable components as well?

maheshwarishikha replied:
added

maheshwarishikha commented:
/run pipeline

rajatagarwal-ibm reviewed (dismissed):
LGTM

ocofaigh commented:
/run pipeline

ocofaigh commented on ibm_catalog.json:
```diff
@@ -0,0 +1,543 @@
+      "label": "[Add-ons Beta] Cloud automation for VPC",
+      "product_kind": "solution",
+      "tags": [
+        "network_vpc",
```
@maheshwarishikha This is a subcategory tag. We are missing a main category tag. I'm going to add the `network` tag or it will prevent us from publishing. FYI you can see all tags by running `ibmcloud catalog filter options --all`

ocofaigh commented on ibm_catalog.json:
```diff
@@ -0,0 +1,543 @@
+        "ibm_created",
+        "target_terraform",
+        "terraform",
+        "support_ibm",
```
@maheshwarishikha I think we need to remove this and add the `ibm_beta` tag. We are not supporting this tile via IBM Cloud - its only experimental beta
